exercises/two-fer/example.exs

```diff
@@ -0,0 +1,11 @@
+defmodule TwoFer do
+  @doc """
+  Two-fer or 2-fer is short for two for one. One for you and one for me.
+  """
+  @spec two_fer(String.t()) :: String.t()
+  def two_fer(name \\ "you")
+  def two_fer(name) when not is_bitstring(name),
+    do: raise ArgumentError, message: "Invalid argument, expected a string as argument!"
```

Comment:
This is great! You can also rely on the FunctionClauseError if you like, which is really nice because it gives you really wonderful error messages showing you which function clauses your arguments did and didn't match. Here's an example: https://cloud.githubusercontent.com/assets/9582/26287340/acd4182a-3e79-11e7-9568-98292e6cbe12.png

That sort of relying on things to fail is a much more common idiom in Elixir than in writing your own exceptions, especially for cases like this where you're dealing with types of arguments. Since that's the common way of doing it, I think we can and should use the tests to nudge folks in that direction, so we can assert that it raises a `FunctionClauseError` instead of asserting that it raises an `ArgumentError` with a specific message.

Reply:
Thanks for the suggestion and example, I didn't know that way!
doing this makes more sense and gets better.